src/robot/variables/tablesetter.py

```diff
@@ -45,23 +47,34 @@ def read(self, variables):
                 var.report_invalid_syntax(err)
 
     def _get_name_and_value(self, name, value, error_reporter):
-        return name[2:-1], VariableTableValue(value, name, error_reporter)
+        name, type_ = split_variable_name_with_bracket_and_type(name)
+        return name[2:-1], VariableTableValue(value, name, type_, error_reporter)
 
 
-def VariableTableValue(value, name, error_reporter=None):
+def VariableTableValue(value, name, type_=None, error_reporter=None):
```

Comment:
I got a feeling it would be simpler if `VariableTableValue` still gets only `name` and then internally checks does it contain type information or not. I might be wrong with this. Seeing how all changes affect each others with such a big PR is hard. As I've written earlier, splitting this to smaller pieces would be great.

Reply:
I split the PR after you request, this part of code is not anymore